### DISCUSSION TITLE
What's your advice on when to adopt the CSF3 format?

### DISCUSSION THREAD

**Discussion by azangru** [storybookjs/storybook#17162] [Help] [answered]
Back in July 2021, @shilman blogged about the new component story format, CSF3. This was also posted in a discussion here: https://github.com/storybookjs/storybook/discussions/15500 Storybook was then at version 6.3, I believe; and I took from the blog post that the new format was likely to be released some time during 6.4.

Now, half a year later, what's your advice on CSF3? Is it still in alpha? Should people start migrating their production code to it? What will happen to the current CSF when CSF3 gets released (I couldn't find in the current docs what version the current CSF is on)?

**Comment by jonniebigodes** [comment, marked as answer]
@azangru thanks for reaching out to us with your questions we really appreciate it, I'll try my best to answer them.

Regarding this one:

> Now, half a year later, what's your advice on CSF3? Is it still in alpha?

It's still a bit experimental and actively being worked on as there are some items that need to be addressed before releasing it.

> Should people start migrating their production code to it?

Probably a bit too soon for that, you could start creating your stories with CSF 3 but probably hold on before putting it into production, and if possible follow up with the team of any issues/bugs you may encounter with that version.

> What will happen to the current CSF when CSF3 gets released (I couldn't find in the current docs what version the current CSF is on)?

To the best of my knowledge, the current version (CSF 2) will be maintained for a bit of time as the community and users will take some time before fully shifting to CSF 3 in their own Storybooks. Our documentation is based on CSF 2.

Let us know if we've answered your questions and please do reach out to us with any further issues and/or questions.

Have a great weekend.

Stay safe

**Reply by azangru** [reply]
Got it 👍  Thank you for the answer.

**Reply by shilman** [reply]
Adding to the excellent answer from @jonniebigodes:

We don't have any concrete plans to deprecate CSF1/2. I think we'll keep backwards compatibility for a few years until CSF3 is widespread and we're happy with all the usage patterns.

I'm a little more bullish on adopting CSF3 in production. We will officially start pushing everybody to migrate in 7.0, once the documentation / auto-migrations / usage patterns (e.g. how to reuse stories in MDX, in other stories, in tests, etc.) are well-established. It's all built into 6.4, without feature flags needed, and I don't anticipate it changing much between now and 7.0 (famous last words).

The back-story is that we were planning to release it in 6.4, but we held back because we don't want to make a minor upgrade any more challenging than it needs to be for users.
